### PR TITLE
Fix ReindexJob

### LINKED ISSUE #958: Fix Spotlight::ReindexJob
Two errors:

```
ActiveRecord::NotNullViolation:
        PG::NotNullViolation: ERROR:  null value in column "on_type" violates not-null constraint
        DETAIL:  Failing row contains (18, null, null, null, null, 5edaaa48-5aa6-47b5-994e-a2750d59cac9, Spotlight::ReindexJob, null, null, completed, null, null, ---
        :progress: 0
        :total: 100
        , 2021-09-21 16:25:47.757731, 2021-09-21 16:25:47.757731).
        : INSERT INTO "spotlight_job_trackers" ("job_id", "job_class", "status", "data", "created_at", "updated_at") VALUES ($1, $2, $3, $4, $5, $6) RETURNING "id"
      # ./spec/jobs/spotlight/reindex_job_spec.rb:29:in `block (2 levels) in <top (required)>'
      # ------------------
      # --- Caused by: ---
      # PG::NotNullViolation:
      #   ERROR:  null value in column "on_type" violates not-null constraint
      #   DETAIL:  Failing row contains (17, null, null, null, null, 5edaaa48-5aa6-47b5-994e-a2750d59cac9, Spotlight::ReindexJob, null, null, in_progress, null, null, --- {}
      #   , 2021-09-21 16:25:47.757731, 2021-09-21 16:25:47.757731).
      #   ./spec/jobs/spotlight/reindex_job_spec.rb:29:in `block (2 levels) in <top (required)>'
```

and

```
16) Spotlight::ReindexJob reindexes an exhibit
      Failure/Error: progress.total = resource_list(job.arguments.first, **pagination).sum(&:estimated_size)
      
      NoMethodError:
        undefined method `estimated_size' for #<ExhibitProxy:0x0000558e7bf53a20>
      # ./app/jobs/spotlight/reindex_job.rb:18:in `block in <class:ReindexJob>'
      # ./spec/jobs/spotlight/reindex_job_spec.rb:21:in `block (2 levels) in <top (required)>'
```

### PR DESCRIPTION
**Pull request by tpendragon**
Closes #958 

Blocked until #961 is merged.

### CODE REVIEW

File: app/jobs/spotlight/reindex_job.rb [hunk 1]
```diff
@@ -7,7 +7,11 @@ class ReindexJob < Spotlight::ApplicationJob
     queue_as :default
 
     include Spotlight::JobTracking
-    with_job_tracking(resource: ->(job) { job.exhibit })
+    # Princeton Update
+    # We have tests that save resources that have no exhibit. If there's no
+    # exhibit, this would error.
```
Comment (hackartisan) [inline app/jobs/spotlight/reindex_job.rb:12]: We don't have use cases for resources with no exhibit, though, right? Should we write a ticket to update the tests?

File: app/models/exhibit_proxy.rb [hunk 2]
```diff
@@ -8,13 +8,17 @@ def initialize(exhibit)
     @exhibit = exhibit
   end
 
-  def reindex(log_entry = nil)
+  def reindex(*_args)
```
Comment (hackartisan) [inline app/models/exhibit_proxy.rb:11]: Will the caller send unpredictable args?

Comment (tpendragon) [reply app/models/exhibit_proxy.rb:11]: It's the same signature as whatever is sent to Spotlight::Resource - we just don't use them, so it seemed like it'd make this class churn less to just accept them and not do anything with them.

File: app/jobs/spotlight/reindex_job.rb (hunk 1 above)
Comment (tpendragon) [reply app/jobs/spotlight/reindex_job.rb:12]: I'm not sure - I was sort of assuming that the test exists for some reason, but I don't know what it would be. There's a bunch of edge cases sneaking around this code base, I just didn't want to accidentally reintroduce an old bug.
